src/interfaces/HposInterface.js

```diff
@@ -78,7 +91,37 @@ const HposInterface = {
     }
 
     return true
-  }
+  },
+
+  updateSettings: async ({ hostPubKey, deviceName, sshAccess }) => {
+    const settingsResponse = await hposCall({ method: 'get', path: 'config' })()
+
+    // Updating the config endpoint requires the hash of the current config to make sure nothing has changed.
+    const headers = {
+      'X-Hpos-Admin-CAS': await hashString(stringify(settingsResponse))
+    }
+
+    // settingsConfig must contain .admin.{email,public_key}, but may contain other arbitrary
+    // data.
+    const settingsConfig = {
+      ...settingsResponse
+    }
+    if (hostPubKey !== undefined) {
+      settingsConfig.admin.public_key = hostPubKey
+    }
+    if (deviceName !== undefined) {
+      settingsConfig.name = deviceName
+    }
+    if (sshAccess !== undefined) {
+      settingsConfig.holoportos = {
+        sshAccess: sshAccess
+      }
+    }
+
+    await hposCall({ method: 'put', path: 'config', headers })(settingsConfig)
+    // We don't assume the successful PUT /api/v1/config returns the current config
```

Comment:
Why not?

Reply:
This comment is actually carried over from a long time ago. Something to do with the way HPOS handled this request at the time we were making hp-admin, but it's worth checking out. @alastairong, is it safe to assume `PUT /api/v1/config` returns the updated config?

Reply:
No it just returns `200`